streams/src/main/java/org/apache/kafka/streams/kstream/Joined.java

```diff
@@ -189,7 +221,22 @@ public Joined<K, V, VO> withOtherValueSerde(final Serde<VO> otherValueSerde) {
      */
     @Override
     public Joined<K, V, VO> withName(final String name) {
-        return new Joined<>(keySerde, valueSerde, otherValueSerde, name);
+        return new Joined<>(keySerde, valueSerde, otherValueSerde, name, gracePeriod);
+    }
+
+    /**
+     * Set the grace period on the stream side of the join. Records will enter a buffer before being processed. Out of order records in the grace period will be processed in timestamp order. Late records, out of the grace period, will be executed right as they come in, if it is past the table history retention this could result in joins on the wrong version or a null join. Long gaps in stream side arriving records will cause records to be delayed in processing, even resulting in be processed out of the grace period window.
```

Comment:
> if it is past the table history retention this could result in joins on the wrong version or a null join

I think this should be "if it is past the table history retention this could result in a null join" instead? I don't think it's possible to join with the wrong version, the only possible issue is that it joins with a null instead.

> even resulting in be processed out of the grace period window

What does this mean?

Reply:
updated and clarified